src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -1209,4 +1219,36 @@ static Credentials createCredentials(
   Downloader getRemoteDownloader() {
     return remoteDownloader;
   }
+
+  private static final class RemoteImportantOutputHandler implements ImportantOutputHandler {
+    @Override
+    public LostArtifacts processOutputsAndGetLostArtifacts(
+        Iterable<Artifact> outputs,
+        ArtifactExpander expander,
+        InputMetadataProvider metadataProvider,
+        ImmutableMap<String, ActionInput> knownLostOutputs) {
+      return new LostArtifacts(
+          knownLostOutputs, new ActionInputDepOwnerMap(knownLostOutputs.values()));
```

Comment:
Does the owner map matter at all for full build rewinding or only action rewinding? It looks like you're just creating an empty one and not populating any owners.

Reply:
It doesn't matter for build rewinding, but as I only noticed after I added more tests, it does for action rewinding for tree artifacts. I am now tracking owners, PTAL.

Reply:
I am working on some refactoring that will hopefully simplify this ownership tracking. I sent the first change for review today: https://bazel-review.googlesource.com/c/bazel/+/274110. The goal is to eventually end up at a place where the ownership tracking is optional (`ActionRewindStrategy` fills it in if it's not present).

Reply:
Thanks! I did wonder why ownership needed to be supplied manually. The CL description helped me understand this better.